tests/e2e/vms/xsvm.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package vms
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/tests"
+	"github.com/ava-labs/avalanchego/tests/fixture/e2e"
+	"github.com/ava-labs/avalanchego/tests/fixture/tmpnet"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/cmd/issue/export"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/cmd/issue/importtx"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/cmd/issue/transfer"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/genesis"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+)
+
+var (
+	subnetAName = "xsvm-a"
+	subnetBName = "xsvm-b"
+)
+
+func XSVMSubnets(nodes ...*tmpnet.Node) []*tmpnet.Subnet {
+	return []*tmpnet.Subnet{
+		newXSVMSubnet(subnetAName, nodes...),
+		newXSVMSubnet(subnetBName, nodes...),
+	}
+}
+
+var _ = ginkgo.Describe("[XSVM]", func() {
+	require := require.New(ginkgo.GinkgoT())
+
+	ginkgo.It("should support transfers between subnets", func() {
+		network := e2e.Env.GetNetwork()
+
+		pluginDir, err := network.DefaultFlags.GetStringVal(config.PluginDirKey)
+		require.NoError(err)
+		const xsvmPluginFilename = "v3m4wPxaHpvGr8qfMeyK6PRW3idZrPHmYcMTt7oXdK47yurVH"
+		xsvmPluginPath := filepath.Join(pluginDir, xsvmPluginFilename)
+		ginkgo.By("checking that xsvm plugin binary exists at path "+xsvmPluginPath, func() {
+			_, err := os.Stat(xsvmPluginPath)
+			require.NoError(err)
+		})
+
+		sourceSubnet := network.GetSubnet(subnetAName)
+		require.NotNil(sourceSubnet)
+		destinationSubnet := network.GetSubnet(subnetBName)
+		require.NotNil(destinationSubnet)
+
+		sourceChain := sourceSubnet.Chains[0]
+		destinationChain := destinationSubnet.Chains[0]
+
+		apiNode := network.Nodes[0]
+		tests.Outf(" issuing transactions on %s (%s)\n", apiNode.NodeID, apiNode.URI)
+
+		destinationKey, err := secp256k1.NewPrivateKey()
+		require.NoError(err)
+
+		ginkgo.By(fmt.Sprintf("exporting from chain %s on subnet %s", sourceChain.ChainID, sourceSubnet.SubnetID))
+		exportTxStatus, err := export.Export(
+			e2e.DefaultContext(),
+			&export.Config{
+				URI:                apiNode.URI,
+				SourceChainID:      sourceChain.ChainID,
+				DestinationChainID: destinationChain.ChainID,
+				Amount:             units.Schmeckle,
+				To:                 destinationKey.Address(),
+				PrivateKey:         sourceChain.PreFundedKey,
+			},
+		)
+		require.NoError(err)
+		tests.Outf(" issued transaction with ID: %s\n", exportTxStatus.TxID)
+
+		ginkgo.By(fmt.Sprintf("issuing transactions on chain %s on subnet %s to activate snowman++ consensus",
+			destinationChain.ChainID, destinationSubnet.SubnetID))
```

Comment:
this is because destinationChain has only genesis right now, and we need at least one block to activate snowman++ fork. Do I understand correctly?